### PR TITLE
Make Authenticated Calls to GitHub

### LINKED ISSUE #167997: [Feature Request]: Support PAT to avoid "API rate limit exceeded"
### Description of the new feature/enhancement

I'm currently using `Tools/SandboxTest.ps1` for testing manifests locally. Sometimes I will trigger an error as follows.

> [!NOTE]
>
> Real IP address is masked for privacy. Just use `127.0.0.1` as a placeholder.

```text
Invoke-RestMethod: C:\Repository\winget-pkgs\Tools\SandboxTest.ps1:86:26
Line |
  86 |  … IResponse = Invoke-RestMethod 'https://api.github.com/repos/microsoft …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     |  {   "message": "API rate limit exceeded for 127.0.0.1. (But here\u0027s the good news: Authenticated requests get a higher rate limit. Check out the documentation
     | for more details.)",   "documentation_url": "https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting" }
```

### Proposed technical implementation details (optional)

Upgrade `Tools/SandboxTest.ps1` and other Powershell scripts, so they can support loading PAT from environment variables or _Windows Credential Manager_. PAT authentication can get grant higher rate limitations for developers and avoid seeing this "limit exceeded" error.

```[tasklist]
### Tasks
- [ ] #168000
- [ ] https://github.com/microsoft/winget-pkgs/pull/168384
```

### PR DESCRIPTION
**Pull request by Trenly**
Checklist for Pull Requests
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Is there a linked Issue?
  * Resolves #168000 
  * Resolves #167997

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/168384)

### CODE REVIEW

File: Tools/SandboxTest.ps1 [hunk 1]
```diff
@@ -82,8 +82,59 @@ New-Item $tempFolder -ItemType Directory -ErrorAction SilentlyContinue | Out-Nul
...
+  $requestParameters = @{
+    Uri = 'https://api.github.com/repos/microsoft/winget-cli/releases?per_page=100'
+  }
+
+  if (Test-GitHubToken) {
+    $requestParameters.Add('Authentication', 'Bearer')
+    $requestParameters.Add('Token', $(ConvertTo-SecureString "$env:GITHUB_TOKEN" -AsPlainText))
+  } else {
+    # See https://docs.github.com/zh/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#primary-rate-limit-for-unauthenticated-users
+    Write-Host "WARNING: You may encounter 'API rate limit exeeded' error! Please consider adding `GITHUB_TOKEN` to your environment variable."
```
Comment (AsciiWolf) [inline Tools/SandboxTest.ps1:130]:
```suggestion
    Write-Host "WARNING: You may encounter 'API rate limit exceeded' error! Please consider adding `GITHUB_TOKEN` to your environment variable."
```

Small typo.

**Review by AsciiWolf — CHANGES_REQUESTED** [review]

**Comment by stephengillie** [comment]
>   # The rate limit for authorized users is usually much higher than 60
It's usually closer to 5000:

```PowerShell
PS C:\ManVal> Get-GitHubRateLimit

Wednesday, August 21, 2024 12:50:56

source    : Unlogged
limit     : 60
used      : 2
remaining : 58
reset     : 8/21/2024 12:32:29


source    : Logged
limit     : 5000
used      : 42
remaining : 4958
reset     : 8/21/2024 12:31:39
```

**Review by stephengillie — CHANGES_REQUESTED** [review]
This change would allow the token to be specified, instead of hard-coded. It would allow for alternate token storage locations, as well as a useful tool to validate other/random tokens.

I currently store my GitHub token in a variable in my $Profile, not in an environment variable. This change would allow this tool to work for my use case also.

**Comment by Trenly** [comment]
@stephengillie - Just adding it there wouldn’t work, it would have to also be added as a parameter of the entire script, which then needs to be validated and passed through to this sub-function.

**Review by stephengillie — APPROVED** [review]

**Comment by Dragon1573** [comment]
Sorry @PckgrBot, why this pull request reverted?

**Pull request merged**